python/Ganga/GPIDev/Lib/File/MassStorageFile.py

```diff
@@ -107,7 +111,36 @@ def _on_attribute__set__(self, obj_type, attrib_name):
     def __repr__(self):
         """Get the representation of the file."""
 
-        return "MassStorageFile(namePattern='%s')" % self.namePattern
+        return "%s(namePattern='%s')" % (_getName(self), self.namePattern)
+
+    def mass_line_processor(self, line):
+        lineParts = line.split(' ')
+        pattern = lineParts[1]
+        outputPath = lineParts[2]
+        name = os.path.basename(outputPath).strip('.gz')
```

Comment:
I'm just copying what was there before bugs/warts and all. Is there an easy way to fix this?